### ISSUE TITLE
arm64 ld.lld: warning: stack frame size (2256) exceeds limit (2048) in function 'wlc_phy_txpwr_srom_read_ppr_nphy'

### ISSUE THREAD

**Issue by emojifreak** [ClangBuiltLinux/linux#1554]
All built-in configuration similar to https://github.com/ClangBuiltLinux/linux/issues/1550 gives a build warning at the final stage

```
  LTO     vmlinux.o
ld.lld: warning: stack frame size (2256) exceeds limit (2048) in function 'wlc_phy_txpwr_srom_read_ppr_nphy'
```

This probably causes build failure with `CONFIG_WERROR=y`. `.config` and a reproducing script are attached. It is unclear to me why it is not caught by clang.
[config.txt.gz](https://github.com/ClangBuiltLinux/linux/files/7769480/config.txt.gz)
[clang-build-arch30.sh.gz](https://github.com/ClangBuiltLinux/linux/files/7769481/clang-build-arch30.sh.gz)

**Comment by nickdesaulniers** [comment]
If you run [this](https://github.com/ClangBuiltLinux/frame-larger-than) on the vmlinux, it might be able to help us spot if there's a large stack allocation, or lots of little ones.

**Comment by emojifreak** [comment]
Thank you. I will try to provide further info. by https://github.com/ClangBuiltLinux/frame-larger-than

**Comment by emojifreak** [comment]
@nickdesaulniers It didn't work as

```
ryutaroh@bookworm-amd64:/var/tmp/tmp30/arm64/linux-5.16-rc6$ python3 /var/tmp/frame_larger_than.py drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_n.\
o wlc_phy_txpwr_srom_read_ppr_nphy
failed to parse elf: Magic number does not match
ryutaroh@bookworm-amd64:/var/tmp/tmp30/arm64/linux-5.16-rc6$ file drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_n.o 
drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_n.o: LLVM IR bitcode
```

**Comment by nickdesaulniers** [comment]
@emojifreak right, you're running the script on a .o file which actually is LLVM bitcode (should be a .bc file). Since it's not an ELF object, an ELF parser will fail to find the ELF magic number at the expected location of an ELF object header.  That's why I suggested running it on vmlinux, which will take longer, but is a fully linked ELF object.

**Comment by emojifreak** [comment]
Thanks. The configuration causes this warning fails before making `vmlinux` and I don't have `vmlinux`, so I will seek a configuration building `vmlinux` with this warning... I will report it when I get a useful additional info.

**Comment by nickdesaulniers** [comment]
Does `./scripts/config -d WERROR` allow you to complete the build?

**Comment by emojifreak** [comment]
No, so far. Because I can observe this symptom only with #1550. I need to reproduce this without seeing #1550 so that I can have `vmlinux`...
